addons/knowledge/views/knowledge_views.xml

```diff
@@ -0,0 +1,247 @@
+        <field name="model">knowledge.article</field>
+        <field name="type">form</field>
+        <field name="arch" type="xml">
+            <form class="h-100" js_class="knowledge_view_form">
+                <field name="category" invisible="1"/>
+                <field name="is_locked" invisible="1" readonly="1"/>
+                <field name="user_can_write" invisible="1"/>
+                <div class="d-sm-flex h-100">
+                    <aside class="o_knowledge_aside d-print-none flex-shrink-0 p-0 bg-white">
+                        <div class="d-flex flex-column h-100">
+                            <!-- Search bar -->
+                            <div class="d-flex align-items-center border-bottom bg-light m-3 o_cursor_pointer">
+                                <div class="flex-grow-1 px-2 py-1">Search an article...</div>
+                                <i class="oi oi-search pr-2" title="search"/>
+                            </div>
+                            <!-- File explorer -->
+                            <div class="flex-grow-1 position-relative">
+                                <div class="o_scroll_view px-3">
+                                    <div class="o_knowledge_tree">
+                                        <div class="p-5 text-center">
+                                            <i class="fa fa-circle-o-notch fa-2x fa-spin" title="loader" role="img"/>
+                                        </div>
+                                    </div>
+                                </div>
+                            </div>
+                        </div>
+                    </aside>
+                    <div class="flex-grow-1 position-relative p-0">
+                        <div class="d-flex flex-column h-100">
+                            <!-- Article header -->
+                            <div class="o_knowledge_header d-print-none d-sm-flex flex-row justify-content-between bg-white border-bottom">
+                                <div class="d-flex align-items-center">
+                                    <!-- Breadcrumbs -->
+                                    <ol class="breadcrumb align-items-center p-3" role="navigation">
+                                        <li class="breadcrumb-item d-flex">
+                                            <field attrs="{'readonly': ['|', ('is_locked', '=', True), ('user_can_write', '=', False)]}" class="o_breadcrumb_article_name o_input" name="name"/>
+                                        </li>
+                                    </ol>
+                                    <field name="is_user_favourite" invisible="1"/>
+                                    <i class="fa fa-star o_toggle_favourite" attrs="{'invisible': [('is_user_favourite', '=', False)]}" title="Remove from favourites"/>
+                                    <i class="fa fa-star-o o_toggle_favourite" attrs="{'invisible': [('is_user_favourite', '=', True)]}" title="Add to favourites"/>
```

Comment:
Could this be a regular action in a link by any chance?

```xml
<a name="action_add_favorite" type="object" attrs="{'invisible': [('is_user_favourite', '=', True)]}" 
    class="fa fa-star-o o_toggle_favourite"/>
```

(Did not test this, it's just to show you an example).

EDIT: Or better yet, I see that there is a "boolean_favorite" widget that is supposed to work in form views as well (you have used it for the kanban).